code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -139,6 +139,29 @@
 /mob/living/simple_animal/updatehealth()
 	..()
 	health = CLAMP(health, 0, maxHealth)
+	update_health_hud()
+
+/mob/living/simple_animal/update_health_hud()
+	if(hud_used)
+		var/severity = 0
+		var/healthpercent = (health / maxHealth) * 100
+		switch(healthpercent)
```

Comment:
You don't need switch
Floor of severity/20 should do the trick

Reply:
its a switch because everywhere else its a switch

Reply:
healthpercent/20 *

Reply:
Was following precedent of existing health hud code, the switch is technically more performant

Reply:
Directly using the result of division in the icon state will be more performant than using switch.

Reply:
You will need to clamp it though ad health percentage can be >100% in the case of adminbus or crappy mapping

Reply:
If you want to be extra performant replace division with multiplication 😉